protoc-gen-kotlin/lib/src/commonTest/kotlin/CodeGeneratorTest.kt

```diff
@@ -0,0 +1,143 @@
+package pbandk
+
+import pbandk.gen.CodeGenerator
+import pbandk.gen.File
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class CodeGeneratorTest {
+    @Test
+    fun testOneOfFields_FullyQualifiedSameMessageName() {
+        val stdField = { number: Int, name: String, type: File.Field.Type, fieldName: String, typeName: String ->
+            File.Field.Numbered.Standard(
+                number,
+                name,
+                type,
+                null,
+                false,
+                fieldName,
+                optional = false,
+                packed = false,
+                map = false,
+                kotlinFieldName = fieldName,
+                kotlinLocalTypeName = typeName
+            )
+        }
+
+        val file = File(
+            "sample.proto",
+            "FooBar",
+            "foobar",
+            1,
+            listOf(
+                File.Type.Message(
+                    "Value",
+                    listOf(
+                        File.Field.OneOf(
+                            "Value", listOf(
+                                stdField(1, "int_val", File.Field.Type.INT32, "intVal", "IntVal"),
+                                stdField(2, "str_val", File.Field.Type.STRING, "strVal", "StrVal")
+                            ),
+                            mapOf(
+                                "int_val" to "IntVal",
+                                "str_val" to "StrVal"
+                            ),
+                            "value",
+                            "Value"
+                        )
+                    ),
+                    emptyList(),
+                    false,
+                    "Value"
+                )
+            )
+        )
+
+        val expected = """
```

Comment:
I'm excited to have some tests of the `CodeGenerator`. Those are sorely lacking currently. But I'm a bit worried that this approach (of including the entire generated file as a quoted string) is going to be hard to maintain, especially if the number of tests in this file increases. Plus, it's hard to tell what exactly this test is testing. The real problem was that the generated code in this situation would fail to compile. But this test doesn't actually tell you if the `CodeGenerator`s output will compile.

The way we've handled these kinds of tests in pbandk previously is by creating a `.proto` file with the test case, generating the `.kt` code from that `.proto` file at build time, and making sure that `.kt` file builds successfully as part of the `commonTest` source set. An example of this approach is the `runtime/src/commonTest/proto/pbandk/testpb/test_compiler_limits.proto` file.

P.S. I'd love to replace that approach with a solution that can run in a unit test, if there's a way to do it without the maintenance overhead of the approach you have here.

Reply:
I see your point. Frankly I thought the same that quoted strings are a real hassle to maintain however currently I'm also at a loss for ideas. The compilation check works fine but fails to test for name-related cosmetic fixes. 
In the scope of a unit `CodeGenerator` the write method is protected so can't be unit tested. 

As such, `generate` is the only thing available to us and the generated code itself. However, I'm certainly open to ideas!

Reply:
Testing for name-related cosmetic fixes would be nice (especially to test that generated names don't suddenly change in a new version of pbandk). It seems that's only tangentially related to the bug fix in this PR though, since the bug was not caused by the code generator generating different names compared to before, but rather the bug was caused by the code generator generating names that clashed with each other and failed to compile.

My preference would be to follow the `test_compiler_limits.proto` testing approach for this PR. And tackle tests for name-related cosmetic fixes in a different PR.

By the way, I just remembered that https://github.com/streem/pbandk/issues/48 has an idea for how to run the compiler from unit tests. That could be another approach to try.